crates/re_data_store/src/log_db.rs

```diff
@@ -37,11 +37,21 @@ impl Default for EntityDb {
             data_store: re_arrow_store::DataStore::new(
                 InstanceKey::name(),
                 DataStoreConfig {
-                    component_bucket_size_bytes: 1024 * 1024, // 1 MiB
+                    // Garbage collection of the datastore is currently driven by the `MsgId`
+                    // component column, as a workaround for the `MsgId` mismatch issue.
```

Comment:
What is "the `MsgId` mismatch issue"? Is there an actual issue describing it?

Reply:
Essentially, the store has no easy way of returning a list of `MsgId`s to the main app when it garbage collects data (because there's no such thing as a message in the store: it gets split into an arbitrary number of index and component buckets during insertion).
We do some major hacks in the current GC implementation to somewhat work around the issue, but this leads to a whole bunch of other problems: one of which is addressed by this PR.

Waiting for us to discuss more during this week's session before opening issues.